src/magic_folder/magicfolderdb.py

```diff
@@ -72,9 +75,41 @@
 );
 """
 
+# magic-folder db schema version 2
+SCHEMA_v2 = """
+CREATE TABLE version
+(
+ version INTEGER  -- contains one row, set to 2
+);
+
+CREATE TABLE local_files
+(
+ path                VARCHAR(1024) PRIMARY KEY,   -- UTF-8 filename relative to local magic folder dir
+ size                INTEGER,                     -- ST_SIZE, or NULL if the file has been deleted
+ mtime_ns            INTEGER,                     -- ST_MTIME in nanoseconds
+ ctime_ns            INTEGER,                     -- ST_CTIME in nanoseconds
+ version             INTEGER,
+ last_uploaded_uri   VARCHAR(256),                -- URI:CHK:...
+ last_downloaded_uri VARCHAR(256),                -- URI:CHK:...
+ last_downloaded_timestamp TIMESTAMP,
+ latest_snapshot VARCHAR(256)
+);
+"""
+
+
+ADD_COLUMN = \
+    """
+    ALTER TABLE local_files ADD COLUMN latest_snapshot VARCHAR(256);
+    """
+
+UPDATE_v1_to_v2 = ADD_COLUMN
+
+UPDATERS = {
+    2: UPDATE_v1_to_v2,
+}
```

Comment:
It looks like the upgrade logic is incomplete.  For what it's worth, I have a very rudimentary SQLite3 schema management code in ZKAPAuthorizer.  Of course, there are also many very featureful (and complex) SQL schema management libraries out there as well.

Taking a step back, I wonder if we want to make any schema changes at this point at all though.  I see that various SQL has been adjusted to load this field from and save this field to the database.  However nothing actually uses this value.  I see this as an attempt to break the large task into digestible smaller pieces and I think that's good.  However, it's not clear where this piece goes.  I suppose a future PR could add a new `Uploader` or `Downloader`-alike which makes use of this information.  Those classes won't need a lot of the existing database fields though, as the existing uploader/downloader don't need the new fields.  So sharing the database between two different consumers seems like it adds extra complexity to both old and new.

If you have a vision for where this goes, great!  Can you lay it out somewhere?  Or if you want to talk about constructing such a vision, let's do that.

Reply:
Yes, most of the existing fields in the database would go away as we discussed in the design document. I was trying to make database and uploader changes as well along with this PR and then realized it would get very big and unmanageable to review. 

I think it makes sense to address this in a later PR along with `Uploader` and `Downloader` changes. I will remove these changes for now.